controller/internal/policy/session.go

```diff
@@ -49,10 +54,28 @@ func NewSessionEnforcer(appEnv *env.AppEnv, frequency time.Duration, sessionTime
 
 func (s *SessionEnforcer) Run() error {
 	oldest := time.Now().Add(s.sessionTimeout * -1)
-	query := fmt.Sprintf("updatedAt < datetime(%s)", oldest.UTC().Format(time.RFC3339))
+	query := fmt.Sprintf("updatedAt < datetime(%s) limit %d", oldest.UTC().Format(time.RFC3339), maxDeletePerIteration)
+
+	for i := 0; i < maxIterations; i++ {
+		ids := make([]string, maxDeletePerIteration)
+		err := s.appEnv.GetHandlers().ApiSession.StreamIds(query, func(id string, err error) error {
+			ids = append(ids, id)
+			return nil
+		})
+
+		if err != nil {
+			pfxlog.Logger().Errorf("encountered error querying for API sessions to remove: %v", err)
+			break
+		}
+
+		if len(ids) == 0 {
+			break
+		}
+
+		for _, id := range ids {
```

Comment:
This is still a nested tx. We probably need to expose the tx in the func passed to streamIds.

Reply:
How is it nested? the `StreamIds` bundles the ids up, view tx closed, then delete in separate txs.

Reply:
Ah, I missed that the delete wasn't in the stream ids func anymore. I still think it should be using QueryIds, since you've essentially reimplemented QueryIds using StreamIds :) But problem is fixed.